src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/Modals/TestModal.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {useMemo, useState} from 'react';
+
+import {Select} from 'antd';
+
+import {ExecutorIcon} from '@atoms';
+
+import {Button, Modal, Text} from '@custom-antd';
+
+import usePressEnter from '@hooks/usePressEnter';
+
+import {TestSuiteStepTest} from '@models/test';
+
+import {useAppSelector} from '@redux/hooks';
+import {selectExecutors} from '@redux/reducers/executorsSlice';
+import {getTestExecutorIcon} from '@redux/utils/executorIcon';
+
+import {useGetAllTestsQuery} from '@services/tests';
+
+import {StyledDelayModalContent, StyledOptionWrapper} from '../SettingsTests.styled';
+
+type TestModalProps = {
+  isTestModalVisible: boolean;
+  setIsTestModalVisible: (flag: boolean) => void;
+  addTest: (test: {name: string; type: string}) => void;
+};
+const {Option} = Select;
+
+const TestModal: React.FC<TestModalProps> = props => {
+  const {isTestModalVisible, setIsTestModalVisible, addTest} = props;
+
+  const {data: allTestsList} = useGetAllTestsQuery();
+
+  const executors = useAppSelector(selectExecutors);
+
+  const [testValue, setTestValue] = useState<string | null>(null);
+
+  const onEvent = usePressEnter();
```

Comment:
Alternatively, `form` with `onSubmit` could be used in the modal. It would work much better (as will be fully native), but if it will be too hard to do - it may be like this for now, and we will come back to this after forms globally will be refactored

Reply:
It is a part of the feature, as it's implemented specifically for this feature, but sure - we could do it later 👍